pkg/destroy/alibabacloud/alibabacloud.go

```diff
@@ -1205,6 +1207,10 @@ func (o *ClusterUninstaller) detachPolicy(policyName string, policyType string,
 }
 
 func (o *ClusterUninstaller) deletePrivateZones(logger logrus.FieldLogger) (err error) {
+	if o.PrivateZoneID != "" {
+		return nil
```

Comment:
I believe we were relying on this to delete the installer created records in the private dns zone, so I think new logic would need to be added (I could be wrong). I think we can merge this anyway and create a new bug. I would rather leak records than destroy records unintentionally.